subsystem/kubeapi_test.go

```diff
@@ -120,35 +128,55 @@ func waitForClusterCRDState(
 		}
 		time.Sleep(time.Second)
 	}
-	Expect(cluster.Status.State).Should(Equal(state))
+	Expect(clusterDeployment.Status.Conditions[0].Message).Should(Equal(state))
 	successInARaw++
 	Expect(successInARaw).Should(Equal(minSuccessesInRow))
 }
 
-func getDefaultClusterSpec(secretRef *corev1.SecretReference) *v1alpha1.ClusterSpec {
-	return &v1alpha1.ClusterSpec{
-		Name:                     "test-cluster",
-		OpenshiftVersion:         "4.6",
-		BaseDNSDomain:            "test.domain",
-		ClusterNetworkCidr:       "10.128.0.0/14",
-		ClusterNetworkHostPrefix: 23,
-		ServiceNetworkCidr:       "172.30.0.0/16",
-		IngressVip:               "1.2.3.9",
-		APIVip:                   "1.2.3.8",
-		SSHPublicKey:             sshPublicKey,
-		VIPDhcpAllocation:        false,
-		PullSecretRef:            secretRef,
-		ProvisionRequirements: v1alpha1.ProvisionRequirements{
-			ControlPlaneAgents: 3,
-			WorkerAgents:       0,
-			AgentSelector:      metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}},
+func getDefaultClusterDeploymentSpec(secretRef *corev1.LocalObjectReference) *hivev1.ClusterDeploymentSpec {
+	return &hivev1.ClusterDeploymentSpec{
+		ClusterName: "test-cluster",
+		BaseDomain:  "test.domain",
+		Platform: hivev1.Platform{
+			AgentBareMetal: &agentv1.BareMetalPlatform{
+				APIVIP:            "1.2.3.8",
+				IngressVIP:        "1.2.3.9",
+				VIPDHCPAllocation: agentv1.Disabled,
+			},
+		},
+		PullSecretRef: secretRef,
+		Provisioning: &hivev1.Provisioning{
+			InstallStrategy: &hivev1.InstallStrategy{
+				Agent: &agentv1.InstallStrategy{
+					Networking: agentv1.Networking{
+						MachineNetwork: []agentv1.MachineNetworkEntry{
+							{
+								CIDR: "172.30.0.0/16",
+							},
+						},
+						ClusterNetwork: []agentv1.ClusterNetworkEntry{
+							{
+								CIDR:       "10.128.0.0/14",
+								HostPrefix: 23,
+							},
+						},
+					},
+					SSHPublicKey: sshPublicKey,
+					ProvisionRequirements: agentv1.ProvisionRequirements{
+						ControlPlaneAgents: 3,
+						WorkerAgents:       0,
+					},
+					AgentSelector: metav1.LabelSelector{MatchLabels: map[string]string{"key": "value"}},
+				},
+			},
 		},
 	}
 }
 
 func cleanUP(ctx context.Context, client k8sclient.Client) {
 	Expect(client.DeleteAllOf(ctx, &v1alpha1.Cluster{}, k8sclient.InNamespace(Options.Namespace))).To(BeNil())
```

Comment:
Still needed to delete `v1alpha1.Cluster{}`?

Reply:
Nop